src/txmempool.cpp

```diff
@@ -1475,6 +1496,19 @@ void CTxMemPool::UpdateTransactionsPerSecond()
     nTxPerSec += 1 / nSecondsToAverage; // The amount that the new tx will add to the tx rate
     if (nTxPerSec < 0)
         nTxPerSec = 0;
+
+    // Calculate the peak rate if we've gone more that 1 second beyond the last sample time.
+    // This will give us the finest grain peak rate possible for txns per second.
+    static int64_t nLastSampleTime = GetTimeMillis();
+    int64_t nCurrentSampleTime = GetTimeMillis();
+    if (nCurrentSampleTime > nLastSampleTime + 1000)
```

Comment:
Maybe I'm missing something obvious here, but `nLastSampleTime` and `nCurrentSampleTime` are initialize almost at the same time. How is it possible that the `if` clause will ever evaluate to `true`?

Reply:
nLastSampleSampleTime a static so only gets assigned here one time ...so initialially it will be equal to nCurrentSampleTime, but on the on each iteration nCurrentSampleTime is assigned a new value until it is greater than nLastSampleTime which still has the initial value.  Then only when the if statement evaluates true nLastSampleTime is updated again.

Reply:
ouch you are right indeed somewhat I "forgot" to read static. Sorry for the noise.